src/routes/safe/components/Apps/AddAppForm/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { Text, TextField } from '@gnosis.pm/safe-react-components'
+import React from 'react'
+import styled from 'styled-components'
+
+import AppAgreement from './AppAgreement'
+import AppUrl, { AppInfoUpdater, appUrlResolver } from './AppUrl'
+import SubmitButtonStatus from './SubmitButtonStatus'
+
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import GnoForm from 'src/components/forms/GnoForm'
+import Img from 'src/components/layout/Img'
+import appsIconSvg from 'src/routes/safe/components/Transactions/TxsTable/TxType/assets/appsIcon.svg'
+
+const StyledText = styled(Text)`
+  margin-bottom: 19px;
+`
+
+const StyledTextFileAppName = styled(TextField)`
+  && {
+    width: 335px;
+  }
+`
+
+const AppInfo = styled.div`
+  margin: 36px 0 24px 0;
+
+  img {
+    margin-right: 10px;
+  }
+`
+
+const APP_INFO: SafeApp = {
+  id: undefined,
+  url: '',
+  name: '',
+  iconUrl: appsIconSvg,
+  error: false,
+}
+
+interface AddAppProps {
+  appList: SafeApp[]
+  closeModal: () => void
+  formId: string
+  onAppAdded: (app: SafeApp) => void
+  setIsSubmitDisabled: (status: boolean) => void
+}
+
+const AddApp = ({ appList, closeModal, formId, onAppAdded, setIsSubmitDisabled }: AddAppProps): React.ReactElement => {
+  const [appInfo, setAppInfo] = React.useState<SafeApp>(APP_INFO)
+
+  const initialValues = {
+    appUrl: '',
+    agreement: false,
```

Comment:
this could be defined outside the component, also, lets change the name of `agreement` to `agreementAccepted`